cli/args.go

```diff
@@ -14,7 +14,25 @@ import (
 
 // Parse command line options that are passed in for Terragrunt
 func ParseTerragruntOptions(cliContext *cli.Context) (*options.TerragruntOptions, error) {
-	return parseTerragruntOptionsFromArgs(cliContext.Args())
+	terragruntOptions, err := parseTerragruntOptionsFromArgs(cliContext.Args())
+	if err != nil {
+		return nil, err
+	}
+
+	if cliContext.App.Writer == nil {
+		terragruntOptions.Writer = os.Stdout
+
+	} else {
+		terragruntOptions.Writer = cliContext.App.Writer
+	}
+
+	if cliContext.App.ErrWriter == nil {
+		terragruntOptions.ErrWriter = os.Stderr
+	} else {
+		terragruntOptions.ErrWriter = cliContext.App.ErrWriter
+	}
+
+	return terragruntOptions, nil
```

Comment:
Instead of this special logic here, I'd update the `parseTerragruntOptionsFromArgs` to take two `Writer`s as arguments and include them in the `TerragruntOptions` it returns. No need for the `nil` checks either if you always set the `Writer` and `ErrWriter` args.